tkg/clusterclient/credentials.go

```diff
@@ -443,6 +443,43 @@ func (c *client) GetAzureCredentialsFromSecret() (azureclient.Credentials, error
 	return res, nil
 }
 
+func (c *client) GetAzureCredentialsFromIdentity(identityName string, identityNamespace string) (azureclient.Credentials, error) {
+	var (
+		clientSecretName      string
+		clientSecretNamespace string
+	)
+	res := azureclient.Credentials{}
+
+	azureClusterIdentityList := &capzv1beta1.AzureClusterIdentityList{}
+	err := c.ListResources(azureClusterIdentityList, &crtclient.ListOptions{})
```

Comment:
is there any methods that could filter the listed resources during `ListResources()` api call (like within `&crtclient.ListOptions{}` )? Not sure if there will be timing issues if a large amount of cluster identity exists.

Reply:
Thanks @hongtaiwang ! I switch this behavior from list identity resource to get identity resource, now!